erdjs/src/dapp/walletProvider.ts

```diff
@@ -200,9 +201,9 @@ export class WalletProvider implements IDappProvider {
 
             // We adjust the fields, in order to make them compatible with what the wallet expected
             plainTransaction["data"] = transaction.data.valueOf().toString();
-            plainTransaction["value"] = transaction.value.toDenominated();
-            plainTransaction["gasPrice"] = transaction.gasPrice.toDenominated();
-            
+            plainTransaction["value"] = Balance.eGLD(transaction.value.toString()).toString();
```

Comment:
Isn't transaction.value already in denominated form (it's also a Balance)? And then Balance.eGLD applies the denomination again. Maybe `Balance.fromString` should be used here.

Reply:
Yes, it is. I pushed a commit with the change. Thanks for the catch